src/OAuth2/Encryption/EncryptionInterface.php

```diff
@@ -4,8 +4,8 @@
 
 interface EncryptionInterface
 {
-    public function encode($payload, $key, $algorithm = null);
-    public function decode($payload, $key, $algorithm = null);
+    public function encode($payload, $key, $algorithm = null, $keyId = null);
+    public function decode($payload, $key, $allowedAlgorithms = array());
```

Comment:
I don't see the win in breaking BC by changing the signature here... we should keep this as `null`